File: Houseclub/src/main/java/me/grishka/houseclub/api/methods/UpdateName.java
package me.grishka.houseclub.api.methods;

import me.grishka.houseclub.api.BaseResponse;
import me.grishka.houseclub.api.ClubhouseAPIRequest;

public class UpdateName extends ClubhouseAPIRequest<BaseResponse> {
    public UpdateName(String name) {
        super("POST", "update_name", BaseResponse.class);
        requestBody = new Body(name);
    }

    private static class Body {
        public String name;

        public Body(String name) {
            this.name = name;
        }
    }
}

File: Houseclub/src/main/java/me/grishka/houseclub/api/methods/UpdateName.kt
package me.grishka.houseclub.api.methods

import me.grishka.houseclub.api.BaseResponse
import me.grishka.houseclub.api.ClubhouseAPIRequest

class UpdateName(name: String) : ClubhouseAPIRequest<BaseResponse?>("POST", "update_name", BaseResponse::class.java) {
    private class Body(var name: String)

    init {
        requestBody = Body(name)
    }
}